Library/ViewModels/BackerDashboardViewModel.swift

```diff
@@ -0,0 +1,239 @@
+      )
+      .switchMap { _ in
+        AppEnvironment.current.apiService.fetchUserSelf()
+          .prefix(SignalProducer([AppEnvironment.current.currentUser].compact()))
+          .demoteErrors()
+    }
+
+    self.avatarURL = user.map { URL(string: $0.avatar.large ?? $0.avatar.medium) }
+
+    self.backedButtonTitleText = user
+      .map { user in
+        localizedString(
+          key: "projects_count_backed",
+          defaultValue: "%{projects_count} backed",
+          count: user.stats.backedProjectsCount ?? 0,
+          substitutions: ["projects_count": Format.wholeNumber(user.stats.backedProjectsCount ?? 0)]
+        )
+        .replacingOccurrences(of: " ", with: "\n")
+    }
+
+    self.backerLocationText = user.map { $0.location?.displayableName ?? "" }
+
+    self.backerNameText = user.map { $0.name }
+
+    self.configureBackedProjectsController = self.viewDidLoadProperty.signal
+      .map { .backed }
+
+    self.configureSavedProjectsController = self.viewDidLoadProperty.signal
+      .map { .saved }
+
+    self.savedButtonTitleText = user.map { user in
+      localizedString(
+        key: "projects_count_backed",
+        defaultValue: "%{projects_count} saved",
+        count: user.stats.starredProjectsCount ?? 0,
+        substitutions: ["projects_count": Format.wholeNumber(user.stats.starredProjectsCount ?? 0)]
+      )
+      .replacingOccurrences(of: " ", with: "\n")
+    }
+
+    self.backedProjectsAreHidden = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(false),
+      self.backedProjectsButtonProperty.signal.mapConst(false),
+      self.savedProjectsButtonProperty.signal.mapConst(true)
+    )
+
+    self.savedProjectsAreHidden = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(true),
+      self.backedProjectsButtonProperty.signal.mapConst(true),
+      self.savedProjectsButtonProperty.signal.mapConst(false)
+    )
+
+    let selectedButtonIndex = Signal.merge(
+      self.backedProjectsButtonProperty.signal.mapConst(0),
+      self.savedProjectsButtonProperty.signal.mapConst(1)
+    )
+
+    self.setSelectedButton = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(0),
+      selectedButtonIndex
+      )
+
+    self.pinSelectedIndicatorToPage = Signal.merge(
+      self.backedButtonTitleText.mapConst((0, false)),
+      selectedButtonIndex.map { ($0, true) }.skipRepeats(==)
+    )
+
+    self.goToProject = .empty
+
+    self.goToMessages = self.messagesButtonTappedProperty.signal
+
+    self.goToSettings = self.settingsButtonTappedProperty.signal
+
+    self.scrollToProject = self.transitionedToProjectRowAndTotalProperty.signal.skipNil().map(first)
+
+    self.sortBarIsHidden = self.viewDidLoadProperty.signal.mapConst(true)
+
+    self.embeddedViewTopConstraintConstant = self.sortBarIsHidden
+      .map { $0 ? 0.0 : Styles.grid(2) }
+  }
+
+  private let backedProjectsButtonProperty = MutableProperty()
```

Comment:
this property should prob be `backedProjectsButtonTappedProperty`. looking through some of the vm code i got tripped up on its current name